hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -69,6 +74,61 @@ func filterMutableHostedClusterSpecFields(spec *hyperv1.HostedClusterSpec) {
 	}
 }
 
+// validateStructDeepEqual walks through a struct and compares each entry.  If it comes accross a substruct it
+// recursively calls itself.  Returns a list of immutable field errors generated by any field being changed.
+func validateStructDeepEqual(x reflect.Value, y reflect.Value, path *field.Path, errs field.ErrorList) field.ErrorList {
+	for i := 0; i < x.NumField(); i++ {
+		v1 := x.Field(i)
+		v2 := y.Field(i)
+		json_id := x.Type().Field(i).Tag.Get("json")
+		sep := strings.Split(json_id, ",")
+		if len(sep) > 1 {
+			json_id = sep[0]
+		}
+
+		if v1.Kind() == reflect.Pointer {
+			// If this is a pointer to a struct, dereference before continuing.
+			if v1.Elem().Kind() == reflect.Struct {
+				v1 = v1.Elem()
+				v2 = v2.Elem()
+			}
+		}
```

Comment:
should you also check for slices of structs?

Reply:
yes, very true.  Tricky to do though.